src/ssh/azext_ssh/resource_tag_utils.py

```diff
@@ -0,0 +1,107 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core import azclierror
+from knack import log
+
+logger = log.get_logger(__name__)
+# Checking Port and Resource Tag argument and fetching appropiate tag
+
+
+def _handle_os_type_for_tag(cmd, op_info):
+    tags = {}
+    server_error = False
+
+    # Warning about two parameters for port (Port, Resource Tag)
+    if op_info.port and op_info.resource_tag:
+        logger.warning("Warning: Both --port and --resource-tag arguments were specified."
+                            "The --port option will take precedence and the --resource-tag will be ignored."
+                            "To use the port number from the --resource-tag, please omit the --port argument.")
+        return
+    # Only port is specified, use port
+    if op_info.port:
+        return
+    # Retrieving machine's resource tags
+    if op_info.resource_type.lower() == "microsoft.compute/virtualmachines":
```

Comment:
When possible, it's better to avoid making multiple calls to azure, because external calls can be slow. We already make a call that retrieves the required information when we call handle_target_os_type. Can we change the structure of the code so that we only need to make the call once, take all the information we need from azure (agent version, os type, tags) and handle each one appropriately? The way the code is right now, every time we need to retrieve a new property from the resource in azure we need to make too many changes in the code. We can brainstorm ways that we can make this solution more scalable.

Reply:
My first thought was creating a class (Examples: AzureResourceMachine) that stores the needed infromation and be initialized in the handle os type. It can save thgings as type, machine name, tags, etc.